rviz_aerial_plugins/src/rviz_aerial_plugins/displays/goal3D/goal3D_display.cpp

```diff
@@ -499,17 +518,29 @@ Goal3DDisplay::makeBoxControl(visualization_msgs::msg::InteractiveMarker & msg)
   return msg.controls.back();
 }
 
-void Goal3DDisplay::makeQuadrocopterMarker(const tf2::Vector3 & position)
+void Goal3DDisplay::makeQuadrocopterMarker()
 {
+  server_->clear();
+
+  std::string namespace_marker = std::string(namespace_->currentText().toUtf8().constData());
+
+  if(namespace_marker.empty())
+    return;
+
+  std::string str_test = std::string(namespace_marker) + "/odom";
+  geometry_msgs::msg::TransformStamped transform_callback_result = buffer_->lookupTransform("map", str_test, tf2::TimePoint());
+
   visualization_msgs::msg::InteractiveMarker int_marker;
   int_marker.header.frame_id = "map";
-  int_marker.pose.position.x = position.getX();
-  int_marker.pose.position.y = position.getY();
-  int_marker.pose.position.z = position.getZ();
+  int_marker.pose.position.x = transform_callback_result.transform.translation.x;
+  int_marker.pose.position.y = transform_callback_result.transform.translation.y;
+  int_marker.pose.position.z = 3.0;
```

Comment:
Why is this hardcoded to 3.0?

Reply:
the 3.0 is for setting a specific Z to the interactive marker. To be able to visualize it properly. should we define this in other way?

Reply:
It just stands out as a magic number here that might currently work for our use case but in the future will likely not be appropriate. I like starting it at the drone pose better.